BSP/STM32F413/Src/BSP_PWM.c

```diff
@@ -89,13 +89,13 @@ void BSP_PWM_Init(void){
 
     //This deals with setting up the input pin for the contactor
     //setup the input pin
-    GPIO_InitTypeDef GPIO_C1Init;
-    GPIO_C1Init.GPIO_Pin = GPIO_Pin_1; //input pin is gpio B1
-    GPIO_C1Init.GPIO_Mode = GPIO_Mode_IN;
-    GPIO_C1Init.GPIO_Speed = GPIO_Speed_2MHz;
-    GPIO_C1Init.GPIO_PuPd = GPIO_PuPd_DOWN;
-    GPIO_C1Init.GPIO_OType = GPIO_OType_PP;
-    GPIO_Init(GPIOB, &GPIO_C1Init);
+    GPIO_InitTypeDef GPIO_B1Init;
+    GPIO_B1Init.GPIO_Pin = GPIO_Pin_1; //input pin is gpio B1
+    GPIO_B1Init.GPIO_Mode = GPIO_Mode_IN;
+    GPIO_B1Init.GPIO_Speed = GPIO_Speed_2MHz;
+    GPIO_B1Init.GPIO_PuPd = GPIO_PuPd_UP;
+    GPIO_B1Init.GPIO_OType = GPIO_OType_OD;
```

Comment:
Someone else to check if open drain is the right config for this

Reply:
Wait, it should be pp